scipy/special/orthogonal.py

```diff
@@ -194,21 +194,27 @@ def _gen_roots_and_weights(n, mu0, an_func, bn_func, f, df, symmetrize, mu):
     dy /= np.abs(dy).max()
     w = 1.0 / (fm * dy)
 
+    if log_mu0 is not None:
+        if symmetrize: raise ValueError('cannot set symmetrize=True with log_mu0 not None')
+        
+        log_w = np.log(w) + log_mu0 - np.log(w.sum())
```

Comment:
when I tried this I saw negative w values. but I tried the log trick for all roots (not just roots_jacobi)

Reply:
should be more explicit now: when passing log_weights=True, the user *expects* to get the log of the weights, thus they are allowed to be negative